### PR TITLE
Boost for OpenSUSE

### PR DESCRIPTION
**Pull request by neotinker**
This adds or updates boost related keys for openSUSE.

### CODE REVIEW

**Comment by nuclearsandwich** [comment]
@neotinker can you please include links to the packages in the openSUSE package web index when contributing changes as described in the contribution guide: master/CONTRIBUTING.md#rosdep-rules-contributions

**Comment by neotinker** [comment]
> @neotinker can you please include links to the packages in the openSUSE package web index when contributing changes as described in the contribution guide: master/CONTRIBUTING.md#rosdep-rules-contributions

Sorry.

https://software.opensuse.org/package/libboost_atomic1_66_0
https://software.opensuse.org/package/libboost_chrono1_66_0
https://software.opensuse.org/package/libboost_date_time1_66_0
https://software.opensuse.org/package/libboost_filesystem1_66_0
https://software.opensuse.org/package/libboost_iostreams1_66_0
https://software.opensuse.org/package/libboost_math1_66_0
https://software.opensuse.org/package/libboost_program_options1_66_0
https://software.opensuse.org/package/libboost_python-py2_7-1_66_0
https://software.opensuse.org/package/libboost_python-py3-1_66_0
https://software.opensuse.org/package/libboost_random1_66_0
https://software.opensuse.org/package/libboost_regex1_66_0
https://software.opensuse.org/package/libboost_serialization1_66_0
https://software.opensuse.org/package/libboost_system1_66_0
https://software.opensuse.org/package/libboost_thread1_66_0
https://software.opensuse.org/package/libboost_timer1_66_0
https://build.opensuse.org/package/binaries/openSUSE:Leap:15.2/boost:base/standard

**Review by nuclearsandwich — APPROVED** [review]
Thanks for updating. I've got one question about the big `boost` key but other than that this looks good and I don't consider that question blocking if the next reviewer (or myself on return) are satisfied with the answer.

File: rosdep/base.yaml [hunk 1]
```diff
@@ -376,7 +376,7 @@ boost:
   macports: [boost]
   nixos: [boost]
   openembedded: [boost@openembedded-core]
-  opensuse: [boost-devel]
+  opensuse: [libboost_headers1_66_0-devel,libboost_system1_66_0-devel,libboost_thread1_66_0-devel,libboost_chrono1_66_0-devel,libboost_date_time1_66_0-devel,libboost_serialization1_66_0-devel,libboost_filesystem1_66_0-devel,libboost_program_options1_66_0-devel,libboost_signals1_66_0-devel,libboost_regex1_66_0-devel]
```
[the remark this answers is not in the document]
Comment (neotinker) [reply rosdep/base.yaml:379]: There is no boost pattern or metapackage to force the install of everything. "boost-devel" no longer exists.  Here is the changelog 

* Tue Oct 10 2017 adam.majer@suse.de
- Use multibuild setup - build no-dependency libraries in the
  base package and build the rest of the compiled libraries in
  the main variant. This should speed up bootstrapping.
- boost-devel not built by default anymore.
- libboost_headers-devel now provides boost-devel for legacy
  dependencies. If you need compiled boost libraries depend on
  the current compiled devel subpackage.
- run %fdupes only on the header files and documentation
- drop build dependencies on gcc-fortran, chrpath.

So libboost_headers-devel "provides"  the name  "boost-devel" but it has no requirements for any of the other devel libraries which the original "boost-devel" had to pull in everything.

neotinker@ros-15_2:~> rpm -q libboost_headers1_66_0-devel --provides
boost-devel = 1.66.0
libboost_headers-devel = 1.66.0
libboost_headers1_66_0-devel = 1.66.0-lp152.6.2
libboost_headers1_66_0-devel(x86-64) = 1.66.0-lp152.6.2
neotinker@ros-15_2:~> rpm -q libboost_headers1_66_0-devel --requires
boost-license1_66_0
libstdc++-devel
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PartialHardlinkSets) <= 4.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
rpmlib(PayloadIsXz) <= 5.2-1
neotinker@ros-15_2:~> 

So, if you need a key that pulls in all the libboost packages, then the boost packages all need to be specified individually.

[the remark this answers is not in the document]
Comment (nuclearsandwich) [reply rosdep/base.yaml:379]: Got it. Thanks for elaborating.

**Review by ivanpauno — APPROVED** [review]

**Pull request merged**